### DISCUSSION TITLE
Desktop login fails with Starship enabled

### DISCUSSION THREAD

**Discussion by marcdw1289** [starship/starship#1202] [General]
Did a search first but couldn't find anything related.
For quite some time I couldn't figure out why I wasn't able to login to my desktop via the display manager. It just kept looping. Discovered that the starship line in my .bashrc ( `eval "$(starship init bash)"` ) was the reason. Soon as I comment it out logins work fine. Logging into the console and using startx isn't a problem.
Any ideas on why xdm doesn't like that? Nothing sticks out in the X or xdm logs.

![Screenshot at 2020-05-14 02-59-55](https://user-images.githubusercontent.com/42319182/81921219-1c8bb000-958f-11ea-989f-b2619b97889c.png)

**Comment by sandorex** [comment]
While i am no expert in this, you're probably running starship in login shell too
You should put  `[[ $- != *i* ]] && return` before starship init to make it run only in interactive mode

**Reply by marcdw1289** [reply]
Yep, that worked out perfectly. Thank you very much. :-)
